src/main/java/com/zipcodewilmington/looplabs/IntegerDuplicateDeleter.java

```diff
@@ -5,4 +5,80 @@
  * @ATTENTION_TO_STUDENTS You are forbidden from modifying the signature of this class.
  */
 public final class IntegerDuplicateDeleter extends DuplicateDeleter<Integer> {
+    public IntegerDuplicateDeleter(Integer[] intArray) {
+
+
+        super(intArray);
+    }
+
+    @Override
+    public Integer[] removeDuplicates(int maxNumberOfDuplications) {
+
+        String hold = "";
+        int counter = 0;
+
+        // Goes through first index of Array and check for duplicates until end of array
+
+        for (int i = 0; i < this.array.length; i++) {
+            for (int j = 0; j < this.array.length; j++) {
+
+                if (this.array[j].equals(this.array[i])) {
+                    counter++;
+                }
+            }
+            if (counter < maxNumberOfDuplications) {
+                hold += this.array[i] + " ";
+            }
+            counter = 0;
+        }
+
+        // I know how ugly this looks but....
+        // This says if the returned Array is empty with a space it returns an empty array
+        // If not it returns the original array that gets converted
+        if (hold == "") {
+            String[] emptyArray = new String[0];
+            Integer[] emptyArray2 = new Integer[emptyArray.length];
```

Comment:
a few things. 1) the biggest reason why this is ugly is because the variable names make these seem useless. Why initialize a variable you intend to always be an empty array? Either you don't intend this in which case the name is bad, or you do in which case this code is absolutely wasted.